socorro/processor/mozilla_transform_rules.py

```diff
@@ -590,10 +591,68 @@ def __init__(self, config):
         # NOTE(willkg): These config values come from Processor2015 instance and are
         # used for lookup in product_versions
         database = config.database_class(config)
-        self.transaction = config.transaction_executor_class(
-            config,
-            database,
-        )
+        self.transaction = config.transaction_executor_class(config, database)
+
+        try:
+            sentry_dsn = self.config.sentry.dsn
+        except (AttributeError, KeyError):
+            sentry_dsn = None
+        self.sentry_dsn = sentry_dsn
+
+    def _get_real_version_db2(self, product, channel, build_id):
+        """Return real version number from crashstats_productversion table
+
+        NOTE(willkg): This is not live, yet--we're doing this to see how it
+        performs compared to the existing product_version table.
+
+        :arg product: the product
+        :arg build_id: the build id as a string
+        :arg channel: the release channel
+
+        :returns: ``None`` or the version string that should be used
+        """
+        if not (product and channel and build_id):
+            return None
+
+        # Fix the product so it matches the data in the table
+        if (product, channel) == ('firefox', 'aurora') and build_id > '20170601':
+            product = 'DevEdition'
+        elif product == 'firefox':
+            product = 'Firefox'
+        elif product in ('fennec', 'fennecandroid'):
+            product = 'Fennec'
+
+        sql = """
+            SELECT version_string
+            FROM crashstats_productversion
+            WHERE product_name = %(product)s
+            AND release_channel = %(channel)s
+            AND build_id = %(build_id)s
+        """
+        params = {
+            'product': product,
+            'channel': channel,
+            'build_id': build_id
+        }
+        versions = self.transaction(execute_query_fetchall, sql, params)
```

Comment:
The archivescraper puts the info in the "right place" in regards to channels so this lookup only has to happen once--we don't have to do a second lookup switching things around in certain cases and when the moon is full.